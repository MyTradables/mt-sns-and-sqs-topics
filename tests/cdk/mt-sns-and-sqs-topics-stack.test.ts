import { SynthUtils } from '@aws-cdk/assert';
import { App } from '@aws-cdk/core';
import environmentConfig from '../../bin/stack-config';
import MtSnsAndSqsTopicsStack from '../../lib/mt-sns-and-sqs-topics-stack';

const app = new App();

describe('Stack validation', () => {
  it('should have a correct stack configuration', () => {
    const stack = new MtSnsAndSqsTopicsStack(app, 'mt-sns-and-sqs-topics', environmentConfig);
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});