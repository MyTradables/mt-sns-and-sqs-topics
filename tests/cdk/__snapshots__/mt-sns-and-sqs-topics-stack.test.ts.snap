// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack validation should have a correct stack configuration 1`] = `
Object {
  "Resources": Object {
    "mtkmskey5AA08EF4": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "Description": "MyTradables key used for encrypting",
        "EnableKeyRotation": true,
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kms:Create*",
                "kms:Describe*",
                "kms:Enable*",
                "kms:List*",
                "kms:Put*",
                "kms:Update*",
                "kms:Revoke*",
                "kms:Disable*",
                "kms:Get*",
                "kms:Delete*",
                "kms:ScheduleKeyDeletion",
                "kms:CancelKeyDeletion",
                "kms:GenerateDataKey",
                "kms:TagResource",
                "kms:UntagResource",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "mtkmskeyAlias84716001": Object {
      "Properties": Object {
        "AliasName": "alias/mt-kms-key",
        "TargetKeyId": Object {
          "Fn::GetAtt": Array [
            "mtkmskey5AA08EF4",
            "Arn",
          ],
        },
      },
      "Type": "AWS::KMS::Alias",
    },
    "proxyDlQueueC6DB40B5": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "KmsMasterKeyId": Object {
          "Fn::GetAtt": Array [
            "mtkmskey5AA08EF4",
            "Arn",
          ],
        },
        "QueueName": "proxyDlQueue",
        "Tags": Array [
          Object {
            "Key": "proxyTagkey",
            "Value": "proxyTagValue",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "proxyQueueFC18A893": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "KmsMasterKeyId": Object {
          "Fn::GetAtt": Array [
            "mtkmskey5AA08EF4",
            "Arn",
          ],
        },
        "QueueName": "proxyQueue",
        "RedrivePolicy": Object {
          "deadLetterTargetArn": Object {
            "Fn::GetAtt": Array [
              "proxyDlQueueC6DB40B5",
              "Arn",
            ],
          },
          "maxReceiveCount": 3,
        },
        "Tags": Array [
          Object {
            "Key": "proxyTagkey",
            "Value": "proxyTagValue",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
    "proxyTopic0EBDEF0C": Object {
      "Properties": Object {
        "KmsMasterKeyId": Object {
          "Fn::GetAtt": Array [
            "mtkmskey5AA08EF4",
            "Arn",
          ],
        },
        "Tags": Array [
          Object {
            "Key": "proxyTagKey",
            "Value": "proxyTagValue",
          },
        ],
        "TopicName": "proxyTopic",
      },
      "Type": "AWS::SNS::Topic",
    },
  },
}
`;
